src/extension.ts

```diff
@@ -228,6 +228,36 @@ export namespace Openshift {
     }
 
     export namespace Explorer {
+        export const login = async function loginCluster(odo: odoctl.Odo, explorer: explorerFactory.OpenShiftExplorer) {
+            if(await odo.requireLogin()) {
+                const clusterIP = await vscode.window.showInputBox({prompt: "Mention the IP address of the cluster", validateInput: (value:string) => {
+                    if (value.trim().length === 0) {
+                        return 'Invalid IP';
+                    }
+                    // add IP validation code
+                }});
+
+                const loginMethod = await vscode.window.showQuickPick(['Credentials', 'Token'], {placeHolder: 'Do you want to login using'});
+                if(loginMethod === "Credentials") {
+                    let username = await vscode.window.showInputBox({prompt: "Mention the username", validateInput: (value:string) => {
+                        if (value.trim().length === 0) {
+                            return 'Invalid username';
+                        }
+                    }});
+                    if (!username) { return; }
+                    let passwd  = await vscode.window.showInputBox({prompt: "Mention the password"});
+                    odo.execute(`oc login ${clusterIP} -u ${username} -p ${passwd}`).then(()=>{
```

Comment:
Do not forget to filter out password from odo output channel.

Reply:
Currently it shows the command as it is passed. Need to figure out a way to hide that from channel or we can also remove the oc login from the channel. Do you recommend any other way for this ?